docs/usage/PRODUCT_SYNC.md

```diff
@@ -21,51 +34,132 @@ against a [ProductDraft](https://docs.commercetools.com/http-api-projects-produc
-fields set, otherwise they won't be matched.
-
-3. The sync expects all variants of the supplied list of `ProductDraft`s to have their `sku` fields set. Also,
-all the variants in the target project are expected to have the `sku` fields set.
-
-4. Every product may have several references including `product type`, `categories`, `taxCategory`, etc. Variants
-of the product also have prices, where each price also has some references including a reference to the `Type` of its 
-custom fields and a reference to a `channel`. All these referenced resources are matched by their `key`s. Therefore, in 
-order for the sync to resolve the actual ids of those references, those `key`s have to be supplied in the following way:
-    - When syncing from a source commercetools project, you can use [`mapToProductDrafts`](https://commercetools.github.io/commercetools-sync-java/v/3.0.1/com/commercetools/sync/products/utils/ProductReferenceResolutionUtils.html#mapToProductDrafts-java.util.List-)
-     method that maps from a `Product` to `ProductDraft` in order to make them ready for reference resolution by the sync:
-     ````java
-     final List<ProductDraft> productDrafts = ProductReferenceResolutionUtils.mapToProductDrafts(products);
-     ````
-     > Note: Some references in the product like `state`, `customerGroup` of prices, and variant attributes with type `reference` do not support the `ResourceIdentifier` yet, 
-      for those references you need to provide the `key` value on the `id` field of the reference. This means that calling `getId()` on the
-      reference would return its `key`. 
-    
-    - For resolving `key-value-document` (custom object) references on attributes of type `Reference`, `Set` of `Reference`, `NestedType` or `Set` of `NestedType`, The `id` field of the reference in the attribute draft should be defined in the correct format. 
-    The correct format must have a vertical bar `|` character between the values of the container and key.
-    For example, if the custom object has a container value `container` and key value `key`, the `id` field should be `container|key"`,  
-    also, the key and container value should match the pattern `[-_~.a-zA-Z0-9]+`.
+````java
+final SphereClientConfig clientConfig = SphereClientConfig.of("project-key", "client-id", "client-secret");
+
+final SphereClient sphereClient = ClientConfigurationUtils.createClient(clientConfig);
+````
+
+#### Required Fields
+
+The following fields are **required** to be set in, otherwise they won't be matched by sync:
+
+|Draft|Required Fields|Note|
+|---|---|---|
+| `ProductDraft` | `key` |  Also, the products in the target project are expected to have the `key` fields set. | 
+| `ProductVariantDraft`  | `key`, `sku` |  Also, all the variants in the target project are expected to have the `key` and `sku` fields set. | 
+
+#### Reference Resolution 
+
```

Comment:
Besides the description of the approach, can we also explain a little bit why we need Reference Resolution?(e.g. Input draft only provides key, but resource in target project is identified by id, so a key-to-id mapping is available to resolve the reference).

Reply:
I am not sure what you mean here, I think it already explained. How we are implementing it is not needed in the user doc, because it's our concern.

Reply:
But I realize I forgot to add this note: https://github.com/commercetools/commercetools-sync-java/pull/639/commits/3fd17209f8c396eec69739a2ee94b5f557d3084a

Reply:
In the beginning, the document said 

> ...a reference can be created by providing the key instead of the ID with the type [ResourceIdentifier](https://docs.commercetools.com/api/types#resourceidentifier)...

User probably don't understand why there is the key and the ID. We can here explain more, from where the key and ID come. wdyt?

Reply:
The user of sync/developer should have basic CTP knowledge, also here I am giving links to the commercetools docs, so should not be a problem for people who are new to CT.